### PR TITLE
[MINOR] New PR for Support client certificate for artifact resolve

### LINKED ISSUE #367: Support client certificate for artifact resolve
Support TLS client certificate to be used for the SOAP binding of artifact resolve queries.

### PR DESCRIPTION
**Pull request by desjoerd**
I've created a new pull request without all the library and netframework version bumps so this can be a MINOR release with less impact.

This PR should resolve #367 and PR #716 

Please review this and leave comments so I can fix the issues and hopefully get this merged :-).

### CODE REVIEW

**Comment by desjoerd** [comment]
I've fixed the merge conflicts which were a result of moving the tests into a seperate Tests folder. It looks like the automatic build is broken though. (Maybe of the change in the repo Organization).

File: Kentor.AuthServices/Configuration/CertificateElement.cs [hunk 1]
```diff
@@ -109,39 +89,39 @@ public X509Certificate2 LoadCertificate()
...
-                    {
-                        var certs = store.Certificates.Find(X509FindType, FindValue, false);
 
-                        if (certs.Count != 1)
-                        {
-                            throw new InvalidOperationException(
-                                string.Format(CultureInfo.InvariantCulture,
-                                "Finding cert through {0} in {1}:{2} with value {3} matched {4} certificates. A unique match is required.",
-                                X509FindType, StoreLocation, StoreName, FindValue, certs.Count));
-                        }
```
Comment (explunit) [inline Kentor.AuthServices/Configuration/CertificateElement.cs]: Is this logic really changed? How is this related to your feature? I suggest reverting the changes to this entire file.

File: Kentor.AuthServices/WebSSO/Saml2ArtifactBinding.cs [hunk 2]
```diff
@@ -90,22 +90,19 @@ public override UnbindResult Unbind(HttpRequestData request, IOptions options)
...
                 Artifact = artifact,
                 Issuer = options.SPOptions.EntityId
             }.ToXml();
 
-            if (options.SPOptions.SigningServiceCertificate != null)
-            {
-                var xmlDoc = XmlHelpers.XmlDocumentFromString(payload);
-                xmlDoc.Sign(options.SPOptions.SigningServiceCertificate, true);
-                payload = xmlDoc.OuterXml;
-            }
```
Comment (explunit) [inline Kentor.AuthServices/WebSSO/Saml2ArtifactBinding.cs]: What is the significance of moving the payload signing from this location and into `SendSoapRequest`?

File: doc/Configuration.md [hunk 3]
```diff
@@ -771,6 +773,11 @@ Signing | Future | Signing | No
 Encryption | Current | Encryption _unless Future key exists_ then not published | Yes 
 Encryption | Future | Encryption | Yes 
 
+### `<artifactResolutionTlsCertificate>` Element
+*Optional child element of the [`<kentor.authServices>`](#kentorauthservices-section) element.*
+
+The client TLS certificate to add to the artifact resolve SOAP request.
+
```
Comment (explunit) [inline doc/Configuration.md]: Thanks for remembering to update the documentation!

**Comment by desjoerd** [comment]
Thanks for reviewing 👍. I've reverted all those unnecessary changes like you suggested. If something needs to be changed or fixed, let me know ^^.

**Comment by keimpema** [comment]
I need this too. Any change this can be merged soon? I applied this PR to my own repositry successfully. It really is a minor change. The conflicts are easy to solve. 

I am working with my fork (https://github.com/keimpema/Saml2) for now but I prefer to use your nuget package ofc.

Great work btw!
